src/Factory.php

```diff
@@ -13,40 +11,31 @@
 class Factory
 {
     /**
-     * @var Connector
-     */
-    private $connector;
-
-    /**
+     * Creates a memcached client
      * @param LoopInterface $loop
+     * @param string $address
+     * @return Client
      */
-    public function __construct(LoopInterface $loop)
+    public static function createClient(LoopInterface $loop, $address = 'localhost:11211')
     {
-        $this->connector = new Connector($loop);
+        return new Client(self::createConnection($loop, $address),  self::createProtocolParser());
     }
 
     /**
-     * Creates a memcached client connected to a given connection string
-     *
-     * @param string $address Memcached server URI to connect to
-     * @return PromiseInterface resolves with Client or rejects with \RuntimeException
+     * @return Parser
      */
-    public function createClient($address = 'localhost:11211')
+    protected static function createProtocolParser()
     {
-        $promise = $this->connector
-            ->connect($address)
-            ->then(function (ConnectionInterface $stream) {
-                return new Client($stream, $this->createProtocolParser());
-            });
-
-        return $promise;
+        return new Parser(new RequestFactory(), new ResponseFactory());
     }
 
     /**
-     * @return Parser
+     * @param LoopInterface $loop
+     * @param $address
+     * @return Connection
      */
-    private function createProtocolParser()
+    protected static function createConnection(LoopInterface $loop, $address)
     {
-        return new Parser(new RequestFactory(), new ResponseFactory());
+        return new Connection($address, new Connector($loop));
     }
```

Comment:
I'd just hardcode this and accept a `Connector` as constructor argument (if you want).